detectree2/models/train.py

```diff
@@ -433,74 +695,62 @@ def load_json_arr(json_path):
             lines.append(json.loads(line))
     return lines
 
-
-def setup_cfg(base_model:
-              str = "COCO-InstanceSegmentation/mask_rcnn_R_101_FPN_3x.yaml",
-              trains=("trees_train",),
-              tests=("trees_val",),
-              update_model=None,
-              workers=2,
-              ims_per_batch=2,
-              gamma=0.1,
-              backbone_freeze=3,
-              warm_iter=120,
-              momentum=0.9,
-              batch_size_per_im=1024,
-              base_lr=0.001,
-              max_iter=1000,
-              num_classes=1,
-              eval_period=100,
-              out_dir="/content/drive/Shareddrives/detectree2/train_outputs"):
-    """Set up config object
-
-    Args:
-        base_model:
-        trains:
-        tests:
-        update_model:
-        workers:
-        ims_per_batch:
-        gamma:
-        backbone_freeze:
-        warm_iter:
-        momentum:
-        batch_size_per_im:
-        base_lr:
-        max_iter:
-        num_classes:
-        eval_period:
-        out_dir:
-    """
-    cfg = get_cfg()
-    cfg.merge_from_file(model_zoo.get_config_file(base_model))
-    cfg.DATASETS.TRAIN = trains
-    cfg.DATASETS.TEST = tests
-    cfg.DATALOADER.NUM_WORKERS = workers
-    #cfg.SOLVER.IMS_PER_BATCH = ims_per_batch
-    cfg.SOLVER.GAMMA = gamma
-    cfg.MODEL.BACKBONE.FREEZE_AT = backbone_freeze
-    cfg.SOLVER.WARMUP_ITERS = warm_iter
-    cfg.SOLVER.MOMENTUM = momentum
-    #cfg.MODEL.RPN.BATCH_SIZE_PER_IMAGE = batch_size_per_im
-    #cfg.SOLVER.WEIGHT_DECAY = 0.001
-    cfg.SOLVER.BASE_LR = base_lr
-    cfg.OUTPUT_DIR = out_dir
-    os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
-    if update_model is not None:
-        cfg.MODEL.WEIGHTS = update_model
-    else:
-        cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url(base_model)
-
-    cfg.SOLVER.IMS_PER_BATCH = ims_per_batch
-    cfg.SOLVER.BASE_LR = base_lr
-    cfg.SOLVER.MAX_ITER = max_iter
-    cfg.MODEL.ROI_HEADS.NUM_CLASSES = num_classes
-    cfg.TEST.EVAL_PERIOD = eval_period
-    return cfg
-
```

Comment:
the set_up cfg function is used for the new RPN layer and to load its parameters